// @linaria/utils/lib/addIdentifierToLinariaPreval.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.addIdentifierToLinariaPreval = addIdentifierToLinariaPreval;
var _createId = require("./createId");
var _scopeHelpers = require("./scopeHelpers");
function getOrAddLinariaPreval(scope) {
  const rootScope = scope.getProgramParent();
  let object = rootScope.getData('__linariaPreval');
  if (object) {
    return object;
  }
  const prevalExport = {
    type: 'ExpressionStatement',
    expression: {
      type: 'AssignmentExpression',
      operator: '=',
      left: {
        type: 'MemberExpression',
        object: (0, _createId.createId)('exports'),
        property: (0, _createId.createId)('__linariaPreval'),
        computed: false
      },
      right: {
        type: 'ObjectExpression',
        properties: []
      }
    }
  };
  const programPath = rootScope.path;
  const [inserted] = programPath.pushContainer('body', [prevalExport]);
  object = inserted.get('expression.right');
  rootScope.setData('__linariaPreval', object);
  return object;
}
function addIdentifierToLinariaPreval(scope, name) {
  const rootScope = scope.getProgramParent();
  const object = getOrAddLinariaPreval(rootScope);
  const newProperty = {
    type: 'ObjectProperty',
    key: (0, _createId.createId)(name),
    value: (0, _createId.createId)(name),
    computed: false,
    shorthand: false
  };
  const [inserted] = object.pushContainer('properties', [newProperty]);
  (0, _scopeHelpers.reference)(inserted.get('value'));
}
